src/Xamarin.Android.Build.Tasks/Resources/LayoutBinding.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+
+using Android.App;
+using Android.Views;
+
+namespace Xamarin.Android.Design
+{
+	abstract class LayoutBinding
+	{
+		public delegate Java.Lang.Object OnLayoutItemNotFoundHandler (int resourceId, Type expectedViewType);
+
+		Activity boundActivity;
+		View boundView;
+		OnLayoutItemNotFoundHandler onLayoutItemNotFound;
+
+		public abstract int ResourceLayoutID { get; }
+
+		protected LayoutBinding (Activity activity, OnLayoutItemNotFoundHandler onLayoutItemNotFound = null)
+		{
+			boundActivity = activity ?? throw new ArgumentNullException (nameof (activity));
+			this.onLayoutItemNotFound = onLayoutItemNotFound;
+		}
+
+		protected LayoutBinding (View view, OnLayoutItemNotFoundHandler onLayoutItemNotFound = null)
+		{
+			boundView = view ?? throw new ArgumentNullException (nameof (view));
+			this.onLayoutItemNotFound = onLayoutItemNotFound;
+		}
+
+		protected T FindView <T> (int resourceId, ref T cachedField) where T: View
+		{
+			if (cachedField != null)
+				return cachedField;
+
+			T ret;
+			if (boundActivity != null)
+				ret = boundActivity.FindViewById <T> (resourceId);
+			else
+				ret = boundView.FindViewById <T> (resourceId);
+
+			if (ret == null && onLayoutItemNotFound != null)
+				ret = (T)onLayoutItemNotFound (resourceId, typeof (T));
+
+			if (ret == null)
+				throw new global::System.InvalidOperationException ($"View not found (Layout ID: {ResourceLayoutID}; Resource ID: {resourceId})");
+
+			cachedField = ret;
+			return ret;
+		}
+#if __ANDROID_11__
+		protected global::Android.App.Fragment FindFragment (int resourceId, ref global::Android.App.Fragment cachedField)
```

Comment:
What happened to our idea of doing:

```csharp
protected T FindFragment<T> (int resourceId, global::Android.App.Fragment value, ref T cachedField)
    where T : global::Android.App.Fragment
{
    ...
}
```

With usage of:

```csharp
global::Android.App.Fragment __secondary_log_fragment;
public global::Android.App.Fragment secondary_log_fragment => FindFragment (global::Xamarin.Android.Tests.CodeBehindFew.Resource.Id.secondary_log_fragment, __secondary_log_fragment, ref __secondary_log_fragment);
```

The added second parameter allows the compiler to disambiguate `FindFragment()` overloads.

Reply:
I'm in two minds about this... On one hand, it provides better typing for the properties. On second hand, this is not a nice hack. Dunno...

Reply:
How is it not a "nice hack"? Regardless, this API isn't for use by developers; it's for use by our generated code. As such, it could be the ugliest code imaginable and *still* be fine.